From: Emil Tantilov <emil.s.tantilov@intel.com>
Date: Tue, 25 Apr 2017 11:31:06 -0700
Subject: ixgbe: enable L3/L4 filtering for Tx switched packets
Patch-mainline: v4.13-rc1
Git-commit: e6b41c888154b5c529ba4d65b6fc55f2a7ae4d75
References: bsc#1056657 FATE#322189 bsc#1056653 FATE#322190

This will ensure that VF-to-VF traffic on the same PF
is filtered to allow RSS operation.

Signed-off-by: Emil Tantilov <emil.s.tantilov@intel.com>
Tested-by: Andrew Bowers <andrewx.bowers@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Acked-by: Benjamin Poirier <bpoirier@suse.com>
---
 drivers/net/ethernet/intel/ixgbe/ixgbe_main.c |    3 +++
 1 file changed, 3 insertions(+)

--- a/drivers/net/ethernet/intel/ixgbe/ixgbe_main.c
+++ b/drivers/net/ethernet/intel/ixgbe/ixgbe_main.c
@@ -3803,6 +3803,9 @@ static void ixgbe_setup_mrqc(struct ixgb
 				mrqc = IXGBE_MRQC_VMDQRSS32EN;
 			else
 				mrqc = IXGBE_MRQC_VMDQRSS64EN;
+
+			/* Enable L3/L4 for Tx Switched packets */
+			mrqc |= IXGBE_MRQC_L3L4TXSWEN;
 		} else {
 			if (tcs > 4)
 				mrqc = IXGBE_MRQC_RTRSS8TCEN;
